src/EnergyPlus/UnitarySystem.cc

```diff
@@ -10917,6 +10941,83 @@ namespace UnitarySystems {
+    {
+        if (SuppHeatLoad <= 0.0) {
+            this->m_SuppHeatPartLoadFrac = 0.0;
+            this->m_SuppHeatingSpeedRatio = 0.0;
+            this->m_SuppHeatingCycRatio = 0.0;
+            return;
+        }
+        constexpr bool SuppHeatingCoilFlag(true);
+        Real64 PartLoadFrac = 0.0;
+        Real64 SpeedRatio = 0.0;
+        Real64 CycRatio = 0.0;
+        std::string CompName = this->m_SuppHeatCoilName;
+        int CompIndex = this->m_SuppHeatCoilIndex;
+        int FanOpMode = this->m_FanOpMode;
+        Real64 QCoilActual = 0.0; // Heating coil operating capacity [W]
+        int SpeedNum = 0;
+        // Get full load result
+        PartLoadFrac = 1.0;
+        CycRatio = 1.0;
+        SpeedRatio = 1.0;
+        int SolFla = 0.0;
+
+        std::vector<Real64> Par(8); // Parameter array passed to solver
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        int constexpr MaxIte(500);    // Maximum number of iterations for solver
+        Real64 constexpr Acc(1.0e-3); // Accuracy of solver result
+
+        for (SpeedNum = 1; SpeedNum <= this->m_NumOfSpeedSuppHeating; ++SpeedNum) {
+            this->m_SuppHeatingSpeedNum = SpeedNum;
+            HeatingCoils::SimulateHeatingCoilComponents(state,
+                                                        CompName,
+                                                        FirstHVACIteration,
+                                                        SuppHeatLoad,
+                                                        CompIndex,
+                                                        QCoilActual,
+                                                        SuppHeatingCoilFlag,
+                                                        FanOpMode,
+                                                        PartLoadFrac,
+                                                        SpeedNum,
+                                                        SpeedRatio);
+            if (QCoilActual > SuppHeatLoad) break;
+        }
+        if (QCoilActual < SuppHeatLoad) {
+            this->m_SuppHeatPartLoadFrac = 1.0;
+            this->m_SuppHeatingSpeedRatio = 1.0;
+            this->m_SuppHeatingCycRatio = 1.0;
+            this->m_SuppHeatingSpeedNum = this->m_NumOfSpeedSuppHeating;
+            return;
+        } else {
+
+            Par[1] = double(this->m_SuppHeatCoilIndex);
+            Par[2] = SuppHeatLoad;
+            Par[3] = double(this->m_UnitarySysNum);
+            // Par(4) = CycRatio or SpeedRatio
+            Par[5] = this->m_SuppHeatingSpeedNum;
+            Par[6] = double(FanOpMode);
+            Par[7] = static_cast<int>(DataHVACGlobals::CompressorOperation::On); // UnitarySystem(UnitarySysNum)%CompressorOp
+            if (this->m_SuppHeatingSpeedNum > 1.0) {
+                Par[4] = CycRatio;
+                General::SolveRoot(state, Acc, MaxIte, SolFla, SpeedRatio, this->calcMultiStageSuppCoilLoadResidual, 0.0, 1.0, Par);
```

Comment:
It would have been an opportunistic time to try out a different paradigm for the solve root payload, but I won't bog this PR down with that.  Just one more we'll need to convert soon.